plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -526,6 +537,17 @@ function ( array $element ): OD_Element {
 		return $this->elements;
 	}
 
+	/**
+	 * Gets the source of the URL Metric.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return non-empty-string|null Source.
```

Comment:
The literal values could be used instead:

```suggestion
	 * @return 'visitor'|'user'|'synthetic'|null Source.
```